### PR TITLE
Allowing to fix the request base URL via an env var

### PR DESCRIPTION
**Pull request by constantinius**
**Related Issue(s):**

- #581

**Description:**

A new env var `REQUEST_BASE_URL` is introduced to fix the base URL to the supplied value. When this variable is not set, the usual way of determining the base URL is used.

**PR Checklist:**

- [x] `pre-commit` hooks pass locally
- [x] Tests pass (run `make test`)
- [ ] Documentation has been updated to reflect changes, if applicable, and docs build successfully (run `make docs`)
- [ ] Changes are added to the [CHANGELOG](https://github.com/stac-utils/stac-fastapi/blob/main/CHANGES.md).

### CODE REVIEW

File: stac_fastapi/types/tests/test_requests.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+from starlette.requests import Request
+
+from stac_fastapi.types.requests import get_base_url
+
+
+def test_get_base_url_with_env():
+    env = {'REQUEST_BASE_URL': 'http://example.com/path/'}
+    with mock.patch.dict(os.environ, env):
+        print(os.environ)
```
Comment (gadomski) [inline stac_fastapi/types/tests/test_requests.py:12]:
```suggestion
```

File: stac_fastapi/types/stac_fastapi/types/requests.py [hunk 2]
```diff
@@ -1,10 +1,15 @@
...
 from starlette.requests import Request
 
 
 def get_base_url(request: Request) -> str:
     """Get base URL with respect of APIRouter prefix."""
+    if base_url := os.environ.get("REQUEST_BASE_URL"):
+        return base_url
+
     app = request.app
     if not app.state.router_prefix:
```
Comment (gadomski) [inline stac_fastapi/types/stac_fastapi/types/requests.py:14]: Was there a reason to _not_ use the same mechanism as `router_prefix`, i.e. add a new `base_url: Optional[str]` attribute to the app state?

**Comment by vincentsarago** [comment]
As for https://github.com/stac-utils/stac-fastapi/pull/636, I'm not convince we need this. 

If the baseurl is not properly returned by starlette it usually mean there is a configuration issue in the proxy 🤔

**Comment by jonhealy1** [comment]
Should we close this for now?

**Comment by vincentsarago** [comment]
@jonhealy1 IMO yes

**Comment by jonhealy1** [comment]
closing ..... feel free to reopen

**Pull request closed without merging**